
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://looselab.github.io/blog/archive/2020/">
      
      
        <link rel="prev" href="../../">
      
      
        <link rel="next" href="../../../contact/">
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.46">
    
    
      
        <title>2020 - Loose Lab</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.6f8fc17f.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#2020" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Loose Lab" class="md-header__button md-logo" aria-label="Loose Lab" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Loose Lab
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              2020
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/LooseLab" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    LooseLab
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../research/" class="md-tabs__link">
        
  
    
  
  Research

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../people/" class="md-tabs__link">
        
  
    
  
  People

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../media/" class="md-tabs__link">
        
  
    
  
  Media

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../" class="md-tabs__link">
          
  
  Blog

        </a>
      </li>
    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../contact/" class="md-tabs__link">
        
  
    
  
  Contact

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Loose Lab" class="md-nav__button md-logo" aria-label="Loose Lab" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Loose Lab
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/LooseLab" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    LooseLab
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../research/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Research
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../people/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    People
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../media/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Media
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Blog
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Blog
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    
  
  
    <li class="md-nav__item">
      <a href="../../" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Blog
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" checked>
        
          
          <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Archive
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5_2">
            <span class="md-nav__icon md-icon"></span>
            Archive
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
    
      
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    2020
  </span>
  

      </a>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contact/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Contact
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
  <div class="md-content" data-md-component="content">
    <div class="md-content__inner">
      <header class="md-typeset">
        <h1 id="2020">2020</h1>
      </header>
      
        <article class="md-post md-post--excerpt">
  <header class="md-post__header">
    
      <nav class="md-post__authors md-typeset">
        
          <span class="md-author">
            <img src="../../../static/people/default.jpg" alt="Alex Payne">
          </span>
        
      </nav>
    
    <div class="md-post__meta md-meta">
      <ul class="md-meta__list">
        <li class="md-meta__item">
          <time datetime="2020-12-02 00:00:00+00:00">December 2, 2020</time></li>
        
        
          
          <li class="md-meta__item">
            
              5 min read
            
          </li>
        
      </ul>
      
    </div>
  </header>
  <div class="md-post__content md-typeset">
    <h2 id="readfish-multiplatform-update"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/">ReadFish Multiplatform Update</a></h2>
<div style="width: 450px; float: right;">
  <img src="../img/readfish_logo.jpg" width="400px">
</div>

<p>Up until now <a href="https://www.nature.com/articles/s41587-020-00746-x">ReadFish</a>, and selective sequencing, have been limited to users 
with both a linux OS and an NVIDIA GPU. In reality this requires a fairly 
large box - be it a GridION or a desktop able to support an NVIDIA 1080 GPU 
or above.
However, it would be ideal to be able to run selective sequencing on a laptop - 
even if it needs to be a moderately powerful device. This was already possible 
with ReadFish if you have a beefy NVIDIA GPU but many notable laptops <em>cough</em> 
Apple <em>cough</em> don't support this at this time.
Quite quickly after releasing our preprint people suggested using a CPU base 
caller:</p>
<blockquote class="twitter-tweet" data-conversation="none" data-dnt="true"><p lang="en" dir="ltr">Or just do CPU basecalling ;) <a href="https://t.co/ICgitgG4Hj">https://t.co/ICgitgG4Hj</a></p>&mdash; Vlado Boza (@bozavlado) <a href="https://twitter.com/bozavlado/status/1225478731197812737?ref_src=twsrc%5Etfw">February 6, 2020</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>Given the way we developed ReadFish, it is relatively straightforward to use 
different components. We therefore decided to implement an alternative base 
calling strategy using <a href="https://academic.oup.com/bioinformatics/article/36/14/4191/5831289">DeepNano-Blitz</a> instead of <a href="https://www.nanoporetech.com">Guppy</a>. As a 
consequence we have been able to implement selective sequencing on computers 
without GPU and running Linux, macOS and even Windows. This brings the power of 
selective sequencing to laptops (with a few caveats below).</p>
<p>This also enabled <a href="https://twitter.com/mattloose">@mattloose</a> to spend a happy weekend running selective 
sequencing on a human genome at home. </p>
<figure>
  <img src="../img/homeseq.png">
  <figcaption>Happy home readfishing...</figcaption>
</figure>

<h3 id="results"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#results">Results</a></h3>
<p>So far we've run using DeepNano-Blitz on macOS, Windows (WSL), GridION, and 
Linux. In all cases we have seen enrichment comparable to running ReadFish 
using GPU accelerated base calling. A key caveat is that our DeepNano-Blitz 
implementation is currently non-adaptive; that is, it can enrich or deplete 
specific genomes or targets but this implementation does not automatically 
monitor reads as they are written or change targets during the experiment.</p>
<p>With that cleared up - what do we see? 
The figures below show coverage over BRCA1, PML and RARA, for runs on the 
GridION Mk1 CPU, a linux box we found in the lab, a MacBook Pro (2018 - 3.1 
GHz i7), and a windows desktop running Windows Subsystem for Linux (windows 10).
We also show two runs from our paper for comparison - the GridION Mk1 and the 
same linux box using an NVIDIA GTX 1080 GPU.</p>
<p>As can be seen, the coverage obtained with DeepNano-Blitz ReadFish is 
comparable (and on windows better!) than that seen on our original experiments.</p>
<figure>
  <a href="../img/BRCA1.png"><img src="../img/BRCA1.png"></a>
  <figcaption>
    Coverage over BRCA1 when using the COSMIC panel and ReadFish either with 
    Guppy (GPU) base calling or DeepNano-Blitz (cpu). Smaller figures below 
    illustrate coverage over PML and RARA for the same experiments.
  </figcaption>
</figure>

<style>
/* Three image containers (use 25% for four, and 50% for two, etc) */
.column {
  float: left;
  width: 50%;
  padding: 5px;
}

/* Clear floats after image containers */
.row::after {
  content: "";
  clear: both;
  display: table;
}
</style>

<div class="row">
  <div class="column">
    <a href="../img/PML.png"><img src="../img/PML.png"></a>
  </div>
  <div class="column">
    <a href="../img/RARA.png"><img src="../img/RARA.png"></a>
  </div>
</div>

<p>In the plot below you can choose a chromosome and look at the distribution of 
coverage for each target. Broadly speaking this is pretty uniform although a 
few targets have pseudogenes that could be cleaned up in future (for example 
see chromosome 4).</p>
<p>{% include 20201130_plot.html %}</p>
<p>The difference in coverage is largely down to down to yield for each flow cell. 
But there are some subtle differences which become apparent. Compare the GPU 
and CPU coverage on the GridION. The GPU experiment had much lower yield, but 
the median read length was shorter. Similarly, the Linux GPU has very low yield, 
but also one of the best median read lengths we have seen. The take home here 
is that speed is everything for efficient read until experiments. The faster 
you can unblock reads you don't wish to see, the better (and flushing your 
flow cell can help recover capacity). Thus the windows run shown here, at 
17&nbsp;Gb, uses brute force of yield to compensate for the longer reads 
observed. There is still room for tuning the windows based setup. </p>
<table>
<thead>
<tr>
<th style="text-align: right;"></th>
<th>GridION MK1&nbsp;GPU</th>
<th>GridION MK1&nbsp;CPU</th>
<th>Linux GPU</th>
<th>Linux CPU</th>
<th>MacBook Pro&nbsp;CPU</th>
<th>Windows CPU</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: right;"><em>Mean Read Length</em></td>
<td>735.6</td>
<td>878.9</td>
<td>683.7</td>
<td>771.2</td>
<td>1085.0</td>
<td>1146.9</td>
</tr>
<tr>
<td style="text-align: right;"><em>Median Read Length</em></td>
<td>423.0</td>
<td>662.0</td>
<td>402.0</td>
<td>564.0</td>
<td>745.0</td>
<td>823.0</td>
</tr>
<tr>
<td style="text-align: right;"><em>Yield</em></td>
<td>9.08&nbsp;Gb</td>
<td>14.93&nbsp;Gb</td>
<td>5.90&nbsp;Gb</td>
<td>14.31&nbsp;Gb</td>
<td>14.03&nbsp;Gb</td>
<td>17.27&nbsp;Gb</td>
</tr>
<tr>
<td style="text-align: right;"><em>Mean Coverage</em></td>
<td>31.30</td>
<td>29.78</td>
<td>19.11</td>
<td>27.78</td>
<td>29.08</td>
<td>34.47</td>
</tr>
<tr>
<td style="text-align: right;"><em>Coverage Std</em></td>
<td>5.54</td>
<td>5.30</td>
<td>3.23</td>
<td>5.09</td>
<td>5.24</td>
<td>6.62</td>
</tr>
<tr>
<td style="text-align: right;"><em>Flushes</em></td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>3</td>
</tr>
</tbody>
</table>
<h3 id="hardware-requirements"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#hardware-requirements">Hardware Requirements</a></h3>
<p>To run everything from one laptop (or desktop) we recommend at least 16&nbsp;Gb 
of RAM, 8 processing cores and an SSD. The processor needs to be one of intel 
i7 or Xeon. </p>
<p>ReadFish will use four cores for base calling and mapping an entire flow cell 
in real time, leaving the remainder of the CPU power for MinKNOW. We have not 
run on an infinite number of systems and your mileage may vary, but 
DeepNano-Blitz provides a number of different models that you can try.</p>
<h3 id="caveats"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#caveats">Caveats</a></h3>
<p>This branch of ReadFish is under development and  provided as is for you to try 
out. </p>
<p>Currently this is only compatible with R9.4 data on DNA. This contrasts with 
Guppy ReadFish, which can handle RNA as well as DNA and any currently shipping 
flow cell type from ONT.</p>
<p>We have not yet implemented real-time base calling for the downstream reads - 
so any of our tools that look at base called reads to inform the choice of 
reads for sequencing will not currently work.</p>
<h3 id="so-how-do-i-set-this-up"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#so-how-do-i-set-this-up">So how do I set this up?</a></h3>
<p>To get started you will need to download the pre-built DeepNano-Blitz installers:</p>
<ul>
<li>Ubuntu:</li>
<li>Python <a href="/static/deepnano2-0.1-cp36-cp36m-linux_x86_64.whl">3.6</a>, <a href="/static/deepnano2-0.1-cp37-cp37m-linux_x86_64.whl">3.7</a>, <a href="/static/deepnano2-0.1-cp38-cp38-linux_x86_64.whl">3.8</a>, <a href="/static/deepnano2-0.1-cp39-cp39-linux_x86_64.whl">3.9</a>, <a href="/static/deepnano2-0.1-cp310-cp310-linux_x86_64.whl">3.10</a></li>
<li>macOS:</li>
<li>Python <a href="/static/deepnano2-0.1-cp36-cp36m-macosx_10_15_x86_64.whl">3.6</a>, <a href="/static/deepnano2-0.1-cp37-cp37m-macosx_10_15_x86_64.whl">3.7</a>, <a href="/static/deepnano2-0.1-cp38-cp38-macosx_10_15_x86_64.whl">3.8</a>, <a href="/static/deepnano2-0.1-cp39-cp39-macosx_10_15_x86_64.whl">3.9</a>, <a href="/static/deepnano2-0.1-cp310-cp310-macosx_11_0_x86_64.whl">3.10</a></li>
</ul>
<p>If you run into issues accessing or installing these, please <a href="https://github.com/LooseLab/readfish/issues/new/choose">contact us</a>.</p>
<p>On Windows we use Windows Subsytem Linux (WSL) to run ReadFish. 
To setup WSL see the <a href="https://docs.microsoft.com/en-us/windows/wsl/install-win10">Microsoft documentation</a>. Once this is 
installed you may need to install or upgrade the Python3 version 
to be one of 3.6, 3.7, or 3.8; now continue the install using the 
instructions for Linux.</p>
<h5 id="setup-a-virtual-environment"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#setup-a-virtual-environment">Setup a virtual environment:</a></h5>
<pre><code class="language-bash">python3 -m venv venv
source ./venv/bin/activate
pip install --upgrade pip wheel
</code></pre>
<h5 id="install-deepnano-blitz"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#install-deepnano-blitz">Install DeepNano-Blitz:</a></h5>
<p>Choose the <code>.whl</code> file corresponding to your OS and python version. E.g:</p>
<pre><code class="language-bash">pip install deepnano2-0.1-cp36-cp36m-linux_x86_64.whl
</code></pre>
<h5 id="install-readfish"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#install-readfish">Install ReadFish:</a></h5>
<pre><code class="language-bash">pip install git+https://github.com/nanoporetech/read_until_api@v3.0.0
pip install git+https://github.com/LooseLab/readfish@caller_refactor

# Install ont_pyguppy_client_lib that matches 
#   your guppy server version. E.G.
pip install ont_pyguppy_client_lib==4.0.11
</code></pre>
<h5 id="setting-the-cpu-caller"><a class="toclink" href="../../2020/12/02/readfish-multiplatform-update/#setting-the-cpu-caller">Setting the CPU caller</a></h5>
<p>To use DeepNano-Blitz for base calling the <code>caller_settings</code> table in the 
experiment TOML file must contain the <code>network_type</code> parameter. For more 
information on the available parameters see <a href="https://github.com/LooseLab/readfish/blob/caller_refactor/TOML.md#deepnano-blitz">TOML.md</a>.</p>
    
  </div>
</article>
      
        <article class="md-post md-post--excerpt">
  <header class="md-post__header">
    
      <nav class="md-post__authors md-typeset">
        
          <span class="md-author">
            <img src="../../../static/people/matt.jpg" alt="Matt Loose">
          </span>
        
      </nav>
    
    <div class="md-post__meta md-meta">
      <ul class="md-meta__list">
        <li class="md-meta__item">
          <time datetime="2020-11-29 00:00:00+00:00">November 29, 2020</time></li>
        
        
          
          <li class="md-meta__item">
            
              3 min read
            
          </li>
        
      </ul>
      
    </div>
  </header>
  <div class="md-post__content md-typeset">
    <h2 id="introducing-readfish"><a class="toclink" href="../../2020/11/29/introducing-readfish/">Introducing ReadFish</a></h2>
<p><img src="../img/readfish_logo.jpg">
* <a href="http://www.github.com/looselab/readfish">ReadFish</a></p>
<p>Today sees our <a href="https://www.biorxiv.org/content/10.1101/2020.02.03.926956v2">preprint</a> on adaptive sampling graduate to a full paper. This was a lot of work from a lot of people and we're very happy to see the paper out.</p>
<blockquote class="twitter-tweet" data-dnt="true" data-theme="light"><p lang="en" dir="ltr">Read Until to enrich &gt;700 genes to mean 30x coverage on <a href="https://twitter.com/nanopore?ref_src=twsrc%5Etfw">@nanopore</a> single flowcell. Custom panels (up to 25,600 targets so far) as easy as providing a coordinate file. Read more at <a href="https://t.co/aZVOs23PKg">https://t.co/aZVOs23PKg</a> - thanks to <a href="https://twitter.com/alexomics?ref_src=twsrc%5Etfw">@alexomics</a>, <a href="https://twitter.com/Rorymatics?ref_src=twsrc%5Etfw">@Rorymatics</a> and <a href="https://twitter.com/DeepSeqNotts?ref_src=twsrc%5Etfw">@DeepSeqNotts</a> team. <a href="https://t.co/0FBLlGQZdm">pic.twitter.com/0FBLlGQZdm</a></p>&mdash; Matt Loose (@mattloose) <a href="https://twitter.com/mattloose/status/1224350276083240960?ref_src=twsrc%5Etfw">February 3, 2020</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>If you want to read the paper, in Nature Biotechnology, you can do so <a href="https://www.nature.com/articles/s41587-020-00746-x">here</a>.</p>
<p>We show in the final version that we can capture targets at depths in excess of 50x on a single MinION flowcell as can be seen in the figure below.</p>
<figure>
  <a href="../img/50xcov.png"><img src="../img/50xcov.png"></a>
  <figcaption>Illustrating 50x coverage over two genes from the COSMIC panel.</figcaption>
</figure>

<p>This blog is to talk a little about the changes we have made since the pre-print and to explain what we will be doing with our code in the next few months.</p>
<p>Firstly - why the name? Well - you gotta catch 'em all! If you don't know the link between Nanopore and fish then have a look at the software. We were asked to distinguish between our toolset and that provided by Oxford Nanopore - which makes sense given that with the recent release of MinKNOW for GridION Mk1, Oxford Nanopore have shipped an implementation for selecting individual regions from a genome or host enrichment or depletion. This may come to other platforms in the future. </p>
<p>So what does ReadFish do? Well - it enables you to select molecules from a library computationally - to catch whatever you like - on any platform with access to GPU basecalling (currently linux only). We built ReadFish as a platform to develop methods to exploit this unique ability of nanopore sequencers and the current toolkit is just the start. </p>
<figure>
  <a href="../img/covsummary.png"><img src="../img/covsummary.png"></a>
  <figcaption>Coverage over two different runs for the entire COSMIC Panel.</figcaption>
</figure>

<p>To date we have used ReadFish to capture many hundreds of genes to depths of around 30x (and sometimes up to 50x) on a single flowcell, enrich or deplete individual genomes from a background, and a personal favourite, identifying the composition of a sample and selecting individual components dynamically. ReadFish provides the ability to change what you are selecting for during a run in response to the data that has been seen to that point. This makes it adaptive.</p>
<figure>
  <a href="../img/cent_rf.png"><img src="../img/cent_rf.png"></a>
  <figcaption>Proportion of bases sequenced mapping to each genome present within the Zymo collection. However, genomes are identified dynamically by ReadFish using centrifuge during the run.</figcaption>
</figure>

<p>ReadFish currently enables adaptive sampling on any computer able to run GPU basecalling with Guppy. The computer and environment must be powerful enough to basecall at lesast two flowcells in real time and to handle mapping duties. For a human genome scale project, you need at least 16 Gb of RAM. ReadFish will be our vehicle for developing new methods around Read Until so keep an eye on this blog for updates. </p>
<p>Some of our future plans are described in a preprint from De Maio et al <a href="https://www.biorxiv.org/content/10.1101/2020.02.07.938670v2">here</a>.</p>
<p>Advice on how to run ReadFish can be found <a href="https://github.com/looselab/readfish">here</a>.</p>
<p>Please pay particular attention to our instructions to test your experiment before running for real. Playback is your friend...</p>
    
  </div>
</article>
      
        <article class="md-post md-post--excerpt">
  <header class="md-post__header">
    
      <nav class="md-post__authors md-typeset">
        
          <span class="md-author">
            <img src="../../../static/people/matt.jpg" alt="Matt Loose">
          </span>
        
      </nav>
    
    <div class="md-post__meta md-meta">
      <ul class="md-meta__list">
        <li class="md-meta__item">
          <time datetime="2020-11-29 00:00:00+00:00">November 29, 2020</time></li>
        
        
          
          <li class="md-meta__item">
            
              1 min read
            
          </li>
        
      </ul>
      
    </div>
  </header>
  <div class="md-post__content md-typeset">
    <h2 id="paper-using-readfish"><a class="toclink" href="../../2020/11/29/paper-using-readfish/">Paper using ReadFish</a></h2>
<blockquote class="twitter-tweet" data-dnt="true"><p lang="en" dir="ltr">Excited to share our work that demonstrates the clinical potential of targeted long-read sequencing. We use Read Fish (Read Until) on a <a href="https://twitter.com/nanopore?ref_src=twsrc%5Etfw">@nanopore</a> platform to systematically analyze clinical samples and find variants not seen w/clinical testing. 1/n <a href="https://t.co/DjE1Bi7wHh">https://t.co/DjE1Bi7wHh</a></p>&mdash; Danny Miller, MD, PhD (@danrdanny) <a href="https://twitter.com/danrdanny/status/1324079670577410048?ref_src=twsrc%5Etfw">November 4, 2020</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>It's great to see this <a href="https://www.biorxiv.org/content/10.1101/2020.11.03.365395v1">paper</a> from Danny Miller and colleagues who have used ReadFish and selective sequencing to analyse the genomes of 33 individuals. They investigate the utility of adaptive sampling for a range of different questions of relevance to human disease and obtain approximately 20x coverage over all their targets on each flowcell.</p>
<p>This is a really interesting paper and a nice use of selective sequencing.  Now Oxford Nanopore have built this basic use case into MinKNOW, users with access to MinKNOW on a GridION MK1 can run this very easily. If you have access to GPU then you can use our <a href="https://github.com/looslab/readfish">ReadFish</a> tools to do the same thing.</p>
<p>The 20x coverage can be improved by using a nuclease flush of the flowcell every 24 hours or so and also loading slightly more library than you would do normally. Depending on how many reads you will be rejecting, you will alter the read length profile of your library and - effectively - the molarity as far as sequencable ends. So we recommend overloading your library. </p>
    
  </div>
</article>
      
      
        
          



<nav class="md-pagination">
  
</nav>
        
      
    </div>
  </div>

          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs", "navigation.tabs.sticky"], "search": "../../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.83f73b43.min.js"></script>
      
    
  </body>
</html>